### ISSUE TITLE
Would changing a GitHub repository's name break the auth?

### ISSUE THREAD

**Issue by digitalghost-dev** [google-github-actions/auth#292]
### TL;DR

I ran my workflow 4 days ago and it worked just fine. I changed my repository's name two days ago then when I went to run the workflow today, the `'google-github-actions/auth@v1'` step doesn't work now. I have not touched the permissions on the service account.

### Expected behavior

I expected the `'google-github-actions/auth@v1'` step to work to authenticate with Google cloud

### Observed behavior

Received an error stating that permission is denied.

### Action YAML

```yaml
name: Build and Deploy to Cloud Run as Job

on:
  workflow_dispatch:
    inputs:
      logLevel:
        description: 'Log level'
        required: true
        default: 'warning'
        type: choice
        options:
        - info
        - warning
        - debug
  push:
    paths-ignore:
      - '.github/**'
      - 'README.md'
      - '.gitignore'
    branches: [ "main" ]

env:
  PROJECT_ID: ${{ secrets.PROJECT_ID }}
  GAR_LOCATION: us-central1
  JOB: ${{ secrets.JOB }}
  REGISTRY_REPO: ${{ secrets.REGISTRY_REPO }}

jobs:
  deploy:
    permissions:
      contents: 'read'
      id-token: 'write'

    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2
        
      - name: Google Auth
        id: auth
        uses: 'google-github-actions/auth@v1'
        with:
          token_format: 'access_token'
          workload_identity_provider: '${{ secrets.WIF_PROVIDER }}'
          service_account: '${{ secrets.WIF_SERVICE_ACCOUNT }}'
          
      - name: Docker Auth
        id: docker-auth
        uses: 'docker/login-action@v1'
        with:
          username: 'oauth2accesstoken'
          password: '${{ steps.auth.outputs.access_token }}'
          registry: '${{ env.GAR_LOCATION }}-docker.pkg.dev'

      - name: Build and Push Container
        run: |-
          docker build --file ./src/teams/Dockerfile --tag "${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.REGISTRY_REPO }}/pl_teams:1.1" ./
          docker push "${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.REGISTRY_REPO }}/pl_teams:1.1"
      - name: Deploy to Cloud Run
        id: deploy
        uses: 'spencercorwin/deploy-cloudrun@enable-jobs'
        with:
          job: ${{ env.JOB }}
          gcloud_component: beta
          region: ${{ env.GAR_LOCATION }}
          # NOTE: If using a pre-built image, update the image name here
          image: ${{ env.GAR_LOCATION }}-docker.pkg.dev/${{ env.PROJECT_ID }}/${{ env.REGISTRY_REPO }}/premier_league:3.1

      # If required, use the Cloud Run url output in later steps
      - name: Show Output
        run: echo ${{ steps.deploy.outputs.url }}
```
```


### Log output

```text
google-github-actions/auth failed with: retry function failed after 1 attempt: failed to generate Google Cloud access token for ***: (403) {
  "error": {
    "code": 403,
    "message": "Permission 'iam.serviceAccounts.getAccessToken' denied on resource (or it may not exist).",
    "status": "PERMISSION_DENIED",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "IAM_PERMISSION_DENIED",
        "domain": "iam.googleapis.com",
        "metadata": {
          "permission": "iam.serviceAccounts.getAccessToken"
        }
      }
    ]
  }
}
```

**Comment by digitalghost-dev** [comment]
Let me read the troubleshooting steps and report back tomorrow.

**Comment by digitalghost-dev** [comment]
Yep, that was the reason why. I built a new pool and updated the GitHub repository name and it worked. Could this be added to the troubleshooting steps?

**Comment by sethvargo** [comment]
Hi @digitalghost-dev - you could have updated your existing attribute condition to satisfy the constraint, but yes, this is working as intended. In [steps 7 and 8 of the README](https://github.com/google-github-actions/auth/blob/main/README.md#setting-up-workload-identity-federation), you are mapping permissions in GitHub to permissions in GCP; one of the mappings is for repository name. That's not required, and you can map on any value in [GitHub's OIDC token](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect).
